staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_test.go

```diff
@@ -18,133 +18,522 @@ package v1
 
 import (
 	"encoding/json"
-	"reflect"
+	"math"
 	"testing"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/google/go-cmp/cmp"
 )
 
-type JSONSchemaPropsOrBoolHolder struct {
-	JSPoB          JSONSchemaPropsOrBool  `json:"val1"`
-	JSPoBOmitEmpty *JSONSchemaPropsOrBool `json:"val2,omitempty"`
+type marshalTestable interface {
+	json.Marshaler
+	cbor.Marshaler
 }
 
-func TestJSONSchemaPropsOrBoolUnmarshalJSON(t *testing.T) {
-	cases := []struct {
-		input  string
-		result JSONSchemaPropsOrBoolHolder
-	}{
-		{`{}`, JSONSchemaPropsOrBoolHolder{}},
+type marshalTestCase struct {
+	name          string
+	input         marshalTestable
+	wantJSONError bool
+	wantCBORError bool
+	wantJSON      []byte
+	wantCBOR      []byte
+}
 
-		{`{"val1": {}}`, JSONSchemaPropsOrBoolHolder{JSPoB: JSONSchemaPropsOrBool{Allows: true, Schema: &JSONSchemaProps{}}}},
```

Comment:
same question about whether the pre-existing unit tests pass as-is against the non-test changes in this PR?

one way to prove that is to leave the existing unit test methods as-is, only add new tests, show they all pass in a CI run and that the new tests cover all the same test inputs, then delete the old unit test methods (maybe this was already done and I missed the discussion of it earlier)